### DISCUSSION TITLE
Automatically add OCI artifacts type label

### DISCUSSION THREAD

**Discussion by lengrongfu** [goharbor/harbor#17405] [Ideas]
ref: https://github.com/goharbor/harbor/discussions/15057
ref: https://github.com/goharbor/harbor/issues/17395

In harbor 2.6 version, Chartmuseum will be discarded, and the Helm Charts table page will be disabled; when OCI artifacts such as helm and image are pushed to harbor, the type of artifacts cannot be intuitively known.

We discuss how to solve this problem here, and I propose a solution; when pushing artifacts to the harbor, parse the manifest and automatically add a label to display it on the page.

**Comment by qnetter** [comment]
Sorry, no - it is DEPRECATED in 2.6. That means that only catastrophic bugs (bugs that cause Harbor to fail completely) and critical security vulnerabilities (CVSS >=9.0) will be fixed and no new functionality added. The feature will be removed in v2.8.0 or later.

**Reply by lengrongfu** [reply]
maybe I didn't express my needs clearly, I agree to discard Chartmuseum.

as this sentence "As Harbor has been an OCI compatible registry and charts can be managed with OCI compatible way", when we use Harbor to manage charts, We can add labels to charts, as shown below.
- image label
<img width="1156" alt="image" src="https://user-images.githubusercontent.com/15009201/184812680-d7b4ff28-5b32-4260-ac33-20d0c132036f.png">

- chart label
<img width="1175" alt="image" src="https://user-images.githubusercontent.com/15009201/184816805-f71f73fc-acb7-445b-b520-0482b2d3e263.png">

**Comment by AllForNothing** [comment]
@lengrongfu 
Currently you can just use the icon to distinguish them on UI. You can see the type when your mouse hovers on the icon 
In fact, the list API also returns a type property.
![image](https://user-images.githubusercontent.com/30999793/185885488-d90ca4e4-e974-4a91-a158-1df9f86c901a.png)
